From e810a1689e5e405cca0f9218f5e55caeab52ce9d Mon Sep 17 00:00:00 2001
From: Stefan Roese <sr@denx.de>
Date: Thu, 17 Jan 2019 16:43:17 +0100
Subject: [PATCH] MIPS: ralink: dts: gardena_smart_gateway_mt7688: Add
 reset gpio-key

This patch adds the reset GPIO input pin as "gpio-keys" DT node so that
it can be referenced and read from userspace, e.g. via
/dev/input/eventX.

Signed-off-by: Stefan Roese <sr@denx.de>
---
 .../boot/dts/ralink/gardena_smart_gateway_mt7688.dts   | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/arch/mips/boot/dts/ralink/gardena_smart_gateway_mt7688.dts b/arch/mips/boot/dts/ralink/gardena_smart_gateway_mt7688.dts
index c9ae52b3ef56..9689e4d4c953 100644
--- a/arch/mips/boot/dts/ralink/gardena_smart_gateway_mt7688.dts
+++ b/arch/mips/boot/dts/ralink/gardena_smart_gateway_mt7688.dts
@@ -14,6 +14,16 @@
 		reg = <0x0 0x8000000>;
 	};
 
+	gpio-keys {
+		compatible = "gpio-keys";
+
+		user_btn1 {
+			label = "USER_BTN1";
+			gpios = <&gpio0 11 GPIO_ACTIVE_LOW>;
+			linux,code =<KEY_PROG1> ;
+		};
+	};
+
 	leds {
 		compatible = "gpio-leds";
 
-- 
2.20.1

